internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -90,28 +93,84 @@ func (r *ForeignClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 
 	// check if linked advertisement exists
 	if fc.Status.Outgoing.Advertisement != nil {
-		_, err = r.advertisementClient.Resource("advertisements").Get(fc.Status.Outgoing.Advertisement.Name, metav1.GetOptions{})
+		tmp, err = r.advertisementClient.Resource("advertisements").Get(fc.Status.Outgoing.Advertisement.Name, metav1.GetOptions{})
 		if errors.IsNotFound(err) {
 			fc.Status.Outgoing.Advertisement = nil
+			fc.Status.Outgoing.AvailableKubeConfig = false
+			fc.Status.Outgoing.AdvertisementStatus = ""
 			fc.Status.Outgoing.Joined = false
 			fc.Status.Outgoing.RemotePeeringRequestName = ""
 			fc.Spec.Join = false
 			requireUpdate = true
 		}
+
```

Comment:
shouldn't this be in a `else` statement (or, better, in case of the previous error we return)